pkg/mtls/secret_manager.go

```diff
@@ -31,13 +31,18 @@ var (
 	secretManagerInstance = &secretManager{
 		validations: make(map[string]*validation),
 	}
-	sdsCallbacks = []func(*v2.TLSConfig){}
+	sdsCallbacks            = make([]func(*v2.TLSConfig), 0)
+	providerUpdateCallbacks = make([]func(provider types.TLSProvider, config *v2.TLSConfig, info *SecretInfo), 0)
 )
 
 func RegisterSdsCallback(f func(*v2.TLSConfig)) {
 	sdsCallbacks = append(sdsCallbacks, f)
 }
 
+func RegisterProviderUpdateCallbacks(f func(types.TLSProvider, *v2.TLSConfig, *SecretInfo)) {
```

Comment:
could add a comment to explain that callback will be triggered when ca or cert are updated.